tests/components/broadlink/test_config_flow.py

```diff
@@ -0,0 +1,465 @@
+    ):
+        yield
+
+
+async def test_flow_user_works(hass):
+    """Test a config flow initiated by the user.
+
+    Best case scenario with no errors or locks.
+    """
+    device = pick_device(0)
+    mock_device = device.get_mock_api()
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
+
+    with patch("broadlink.discover", return_value=[mock_device]) as mock_discover:
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": device.host, "timeout": device.timeout},
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "finish"
+    assert result["errors"] == {}
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], {"name": device.name},
+    )
+
+    assert result["type"] == "create_entry"
+    assert result["title"] == device.name
+    assert result["data"] == device.get_entry_data()
+
+    assert mock_discover.call_count == 1
+    assert mock_device.auth.call_count == 1
+
+
+async def test_flow_user_already_in_progress(hass):
+    """Test we do not accept more than one config flow per device."""
+    device = pick_device(0)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch("broadlink.discover", return_value=[device.get_mock_api()]):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": device.host, "timeout": device.timeout},
+        )
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch("broadlink.discover", return_value=[device.get_mock_api()]):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": device.host, "timeout": device.timeout},
+        )
+
+    assert result["type"] == "abort"
+    assert result["reason"] == "already_in_progress"
+
+
+async def test_flow_user_device_is_unique(hass):
+    """Test we do not accept more than one config entry per device."""
+    device = pick_device(0)
+    mock_entry = device.get_mock_entry()
+    mock_entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    device.host = "192.168.1.240"  # The IP address has changed.
+    mock_device = device.get_mock_api()
+
+    with patch("broadlink.discover", return_value=[mock_device]):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": device.host},
+        )
+
+    with patch(
+        "homeassistant.components.broadlink.async_unload_entry", return_value=True
```

Comment:
We should check the result here. The result should be abort with reason `"already_configured"`.

We should call `self._abort_if_unique_id_configured` in the config flow after setting the unique_id.

Reply:
Ok. What about reauth? Should I abort or overwrite the entry?

Reply:
The reauth should update an existing entry and abort.

Reply:
Ok.

Reply:
Done.